src/main/fc/stats.c

```diff
@@ -0,0 +1,75 @@
+
+#include "platform.h"
+
+#ifdef USE_STATS
+
+#include "drivers/time.h"
+
+#include "fc/config.h"
+#include "fc/runtime_config.h"
+#include "fc/stats.h"
+#include "pg/stats.h"
+
+#include "io/beeper.h"
+#include "io/gps.h"
+
+
+#define MIN_FLIGHT_TIME_TO_RECORD_STATS_S 10    //prevent recording stats for that short "flights" [s]
+#define STATS_SAVE_DELAY_MS              500    //let disarming complete and save stats after this time
+
+static uint32_t arm_millis;
+static uint32_t arm_distance_cm;
+static uint32_t save_pending_millis;  // 0 = none
+
+#ifdef USE_GPS
+    #define DISTANCE_FLOWN_CM (GPS_distanceFlownInCm)
+#else
+    #define DISTANCE_FLOWN_CM (0)
+#endif
+
+void statsOnArm(void)
+{
+    arm_millis      = millis();
+    arm_distance_cm = DISTANCE_FLOWN_CM;
+}
+
+void statsOnDisarm(void)
+{
+    if (statsConfig()->stats_enabled) {
+        uint32_t dt = (millis() - arm_millis) / 1000;
```

Comment:
Use `timeMs_t` for time in milliseconds.

Reply:
This one keeps seconds, not millis.
OK for others.